---
float32 pos       # res: position
float32 vel
bool success
